core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -336,43 +384,57 @@ public TenantObject setAdapterConfigurations(final List<Map<String, Object>> con
      *                       configured adapters or {@code null} in order to
      *                       remove any existing configuration.
      * @return This tenant for command chaining.
+     * @deprecated Use {@link #setAdapters(List adapters)} instead.
      */
+    @Deprecated
     @JsonIgnore
     public TenantObject setAdapterConfigurations(final JsonArray configurations) {
-        if (configurations == null) {
-            this.adapterConfigurations = null;
-        } else {
-            this.adapterConfigurations = new LinkedList<>();
-            configurations.stream().filter(obj -> JsonObject.class.isInstance(obj)).forEach(config -> {
-                addAdapterConfiguration((JsonObject) config);
-            });
+        this.adapters = null;
+        if (configurations != null) {
+            configurations.stream()
+                    .filter(JsonObject.class::isInstance)
+                    .forEach(config -> addAdapterConfiguration((JsonObject) config));
```

Comment:
this means that we end up with a *half-populated* adapters list if one of the configurations from the JsonArray doesn't contain a *type*, right?

Reply:
yes and to resolve this now I have added a validation to check if the given adapter configurations are valid (i.e being able to map to `Adapter.class` and ensure if the _adapter type_ in the list is unique). Only on successful validation, the given configurations are added.

Reply:
Also new tests have been added to ensure the above.